### DISCUSSION TITLE
Livewire v3 clearing flash data on initial dehydate and not subsequent dehyrdate

### DISCUSSION THREAD

**Discussion by TWithers** [livewire/livewire#7902] [7. Bugs (archived)]
### Livewire version

v3.4

### Laravel version

v10

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Session Flash content is being removed on the initial load of the component, even if it isn't used/needed. This was an issue with v2 (https://github.com/livewire/livewire/issues/2294) but was resolved so that flash data was only removed after subsequent requests/dehydrates, and not the initial dehydate. It looks like this issue was re-introduced with v3.

1. In the controller, flash data: `session()->flash('foo', 'bar');`
2. In the view, before the livewire component: `@php dump(session()->all()); @endphp` [Flash content still there]
3. AFTER the livewire component, in the same view: @php dump(session()->all()); @endphp` [Flash content missing]

### Repository or Wirebox URL

https://wirebox.app/b/x6r67

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@TWithers thanks for reporting! Any chance you could submit a failing test PR?

**Reply by TWithers** [reply]
https://github.com/livewire/livewire/pull/7921
I created a PR with the failing tests. 

I am not sure where to begin solving the issue as I am unsure how Livewire V2 determined subsequent requests. 

My initial thoughts were to add another flashed session variable foreach component on/after render, and only removing the flashed data on the first subsequent request. The only problem with this approach is that the recent changes moved the `session()->forget()` call to the `provide()` static function, and I don't believe there is a way to get the component id/name/key inside the static function.

**Reply by TWithers** [reply]
This would allow all tests to pass... but I am not sure if this is the right way to do it:

SupportRedirects.php
```php
        on('response', function ($response) {
            // If there was no redirect. Clear flash session data.
            if (! static::$atLeastOneComponentHasRedirected && app()->has('session.store')) {
                foreach ($response['components'] as $component) {
                    $snapshot = json_decode($component['snapshot']);
                    $id = $snapshot?->memo?->id;
                    if (! $id || session()->has('_is_subsequent_request_for_'.$id)) {
                        session()->forget(session()->get('_flash.new'));
                        break;
                    } else {
                        session()->flash('_is_subsequent_request_for_'.$id, true);
                    }
                }
            }
        });
```

**Reply by joshhanley** [reply]
@TWithers thanks! Can you please fill out the PR description? Caleb prefers not to have to go digging into discussions when the problem (and possibly solution) can be explained in the PR. Thanks!
